src/main/java/org/htwkvisu/gui/MapCanvas.java

```diff
@@ -116,22 +139,50 @@ public Point2D getCenter() {
         return mapCenter;
     }
 
+    public int getSamplingPixelDensity() {
+        return samplingPixelDensity;
+    }
+
+    public void setSamplingPixelDensity(int samplingPixelDensity) {
+        if (samplingPixelDensity > 0) {
+            this.samplingPixelDensity = samplingPixelDensity;
+        }
+    }
+
     /**
-     * Set the maps scale.
+     * Returns coordinates for samples where to calculate scoring values.
+     * They are positioned in a XY-grid with pixelDensity pixels distance between each sample.
      *
-     * @param scale of map
+     * @param pixelDensity Distance between samples in pixels
+     * @return Sample positions in a grid
      */
-    public void setScale(double scale) {
-        if (scale < ZOOM_MAX) {
-            if (scale > ZOOM_MIN) {
-                this.scale = scale;
-            } else {
-                this.scale = ZOOM_MIN;
+    public ArrayList<ArrayList<Point2D>> getSampleCoordPoints(float pixelDensity) {
```

Comment:
please change the type of matrix to `List<List<Point2D>>` . A `List` is the better type because its the interface of `ArrayList` and in java you should programm against interfaces (more  generally ^^).